### ISSUE TITLE
Firestore Error running ng-demo: "static_assert failed "This hash only works for enumeration types"

### ISSUE THREAD

**Issue by MentalRender** [EddyVerbruggen/nativescript-plugin-firebase#826]
Hi Eddy! Following this steps:

`git clone https://github.com/EddyVerbruggen/nativescript-plugin-firebase.git`

`cd nativescript-plugin-firebase/demo-ng`

`tns run ios`

```
warning: unknown warning option '-Wno-comma' [-Wunknown-warning-option]
1 warning generated.
ld: warning: directory not found for option '-F/Users/mntlrndr/Library/Developer/Xcode/DerivedData/demong-gbcqxlenxtobxxhetxvwluaewvjt/Build/Intermediates/ArchiveIntermediates/demong/BuildProductsPath/Debug-iphoneos/GTMSessionFetcher'
=== BUILD TARGET QBImagePickerController OF PROJECT Pods WITH CONFIGURATION Debug ===
ld: warning: directory not found for option '-F/Users/mntlrndr/Library/Developer/Xcode/DerivedData/demong-gbcqxlenxtobxxhetxvwluaewvjt/Build/Intermediates/ArchiveIntermediates/demong/BuildProductsPath/Debug-iphoneos/BoringSSL'
ld: warning: directory not found for option '-F/Users/mntlrndr/Library/Developer/Xcode/DerivedData/demong-gbcqxlenxtobxxhetxvwluaewvjt/Build/Intermediates/ArchiveIntermediates/demong/BuildProductsPath/Debug-iphoneos/nanopb'
<module-includes>:1:1: warning: umbrella header for module 'grpc' does not include header '/impl/codegen/atm_gcc_sync.h' [-Wincomplete-umbrella]
#import "Headers/grpc.h"
^
<module-includes>:1:1: warning: umbrella header for module 'grpc' does not include header '/impl/codegen/atm_windows.h' [-Wincomplete-umbrella]
<module-includes>:1:1: warning: umbrella header for module 'grpc' does not include header '/impl/codegen/sync_custom.h' [-Wincomplete-umbrella]
<module-includes>:1:1: warning: umbrella header for module 'grpc' does not include header '/impl/codegen/sync_windows.h' [-Wincomplete-umbrella]
<module-includes>:1:1: warning: umbrella header for module 'grpc' does not include header '/support/atm_gcc_atomic.h' [-Wincomplete-umbrella]
<module-includes>:1:1: warning: umbrella header for module 'grpc' does not include header '/support/atm_gcc_sync.h' [-Wincomplete-umbrella]
<module-includes>:1:1: warning: umbrella header for module 'grpc' does not include header '/support/atm_windows.h' [-Wincomplete-umbrella]
<module-includes>:1:1: warning: umbrella header for module 'grpc' does not include header '/support/sync_custom.h' [-Wincomplete-umbrella]
<module-includes>:1:1: warning: umbrella header for module 'grpc' does not include header '/support/sync_posix.h' [-Wincomplete-umbrella]
<module-includes>:1:1: warning: umbrella header for module 'grpc' does not include header '/support/sync_windows.h' [-Wincomplete-umbrella]
10 warnings generated.
10 warnings generated.
=== BUILD TARGET gRPC OF PROJECT Pods WITH CONFIGURATION Debug ===
ld: warning: directory not found for option '-F/Users/mntlrndr/Library/Developer/Xcode/DerivedData/demong-gbcqxlenxtobxxhetxvwluaewvjt/Build/Intermediates/ArchiveIntermediates/demong/BuildProductsPath/Debug-iphoneos/BoringSSL'
ld: warning: directory not found for option '-F/Users/mntlrndr/Library/Developer/Xcode/DerivedData/demong-gbcqxlenxtobxxhetxvwluaewvjt/Build/Intermediates/ArchiveIntermediates/demong/BuildProductsPath/Debug-iphoneos/gRPC-Core'
ld: warning: directory not found for option '-F/Users/mntlrndr/Library/Developer/Xcode/DerivedData/demong-gbcqxlenxtobxxhetxvwluaewvjt/Build/Intermediates/ArchiveIntermediates/demong/BuildProductsPath/Debug-iphoneos/gRPC-RxLibrary'
ld: warning: directory not found for option '-F/Users/mntlrndr/Library/Developer/Xcode/DerivedData/demong-gbcqxlenxtobxxhetxvwluaewvjt/Build/Intermediates/ArchiveIntermediates/demong/BuildProductsPath/Debug-iphoneos/nanopb'
=== BUILD TARGET gRPC-ProtoRPC OF PROJECT Pods WITH CONFIGURATION Debug ===
ld: warning: directory not found for option '-F/Users/mntlrndr/Library/Developer/Xcode/DerivedData/demong-gbcqxlenxtobxxhetxvwluaewvjt/Build/Intermediates/ArchiveIntermediates/demong/BuildProductsPath/Debug-iphoneos/BoringSSL'
ld: warning: directory not found for option '-F/Users/mntlrndr/Library/Developer/Xcode/DerivedData/demong-gbcqxlenxtobxxhetxvwluaewvjt/Build/Intermediates/ArchiveIntermediates/demong/BuildProductsPath/Debug-iphoneos/Protobuf'
ld: warning: directory not found for option '-F/Users/mntlrndr/Library/Developer/Xcode/DerivedData/demong-gbcqxlenxtobxxhetxvwluaewvjt/Build/Intermediates/ArchiveIntermediates/demong/BuildProductsPath/Debug-iphoneos/gRPC'
ld: warning: directory not found for option '-F/Users/mntlrndr/Library/Developer/Xcode/DerivedData/demong-gbcqxlenxtobxxhetxvwluaewvjt/Build/Intermediates/ArchiveIntermediates/demong/BuildProductsPath/Debug-iphoneos/gRPC-Core'
ld: warning: directory not found for option '-F/Users/mntlrndr/Library/Developer/Xcode/DerivedData/demong-gbcqxlenxtobxxhetxvwluaewvjt/Build/Intermediates/ArchiveIntermediates/demong/BuildProductsPath/Debug-iphoneos/gRPC-RxLibrary'
ld: warning: directory not found for option '-F/Users/mntlrndr/Library/Developer/Xcode/DerivedData/demong-gbcqxlenxtobxxhetxvwluaewvjt/Build/Intermediates/ArchiveIntermediates/demong/BuildProductsPath/Debug-iphoneos/nanopb'
warning: unknown warning option '-Wno-comma' [-Wunknown-warning-option]
1 warning generated.
warning: unknown warning option '-Wno-comma' [-Wunknown-warning-option]
1 warning generated.
In file included from /Users/mntlrndr/temp/borrable_sin_mas/nativescript-plugin-firebase/demo-ng/platforms/ios/Pods/FirebaseFirestore/Firestore/core/src/firebase/firestore/model/document.cc:17:
In file included from /Users/mntlrndr/temp/borrable_sin_mas/nativescript-plugin-firebase/demo-ng/platforms/ios/Pods/FirebaseFirestore/Firestore/core/src/firebase/firestore/model/document.h:21:
In file included from /Users/mntlrndr/temp/borrable_sin_mas/nativescript-plugin-firebase/demo-ng/platforms/ios/Pods/FirebaseFirestore/Firestore/core/src/firebase/firestore/model/field_value.h:21:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/map:446:
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/functional:2480:5: error: static_assert failed "This hash only works for enumeration types"
    static_assert(is_enum<_Tp>::value, "This hash only works for enumeration types");
    ^             ~~~~~~~~~~~~~~~~~~~
In file included from /Users/mntlrndr/temp/borrable_sin_mas/nativescript-plugin-firebase/demo-ng/platforms/ios/Pods/FirebaseFirestore/Firestore/core/src/firebase/firestore/model/document.cc:17:
In file included from /Users/mntlrndr/temp/borrable_sin_mas/nativescript-plugin-firebase/demo-ng/platforms/ios/Pods/FirebaseFirestore/Firestore/core/src/firebase/firestore/model/document.h:20:
In file included from /Users/mntlrndr/temp/borrable_sin_mas/nativescript-plugin-firebase/demo-ng/platforms/ios/Pods/FirebaseFirestore/Firestore/core/src/firebase/firestore/model/field_path.h:24:
In file included from /Users/mntlrndr/temp/borrable_sin_mas/nativescript-plugin-firebase/demo-ng/platforms/ios/Pods/FirebaseFirestore/Firestore/core/src/firebase/firestore/model/base_path.h:28:
/Users/mntlrndr/temp/borrable_sin_mas/nativescript-plugin-firebase/demo-ng/platforms/ios/Pods/FirebaseFirestore/Firestore/core/src/firebase/firestore/util/hashing.h:104:17: note: in instantiation of
      template class 'std::__1::hash<firebase::firestore::model::ResourcePath>' requested here
    -> decltype(std::hash<K>{}(value)) {
                ^
/Users/mntlrndr/temp/borrable_sin_mas/nativescript-plugin-firebase/demo-ng/platforms/ios/Pods/FirebaseFirestore/Firestore/core/src/firebase/firestore/util/hashing.h:127:10: note: while substituting
      deduced template arguments into function template 'RankedInvokeHash' [with K = firebase::firestore::model::ResourcePath]
  return RankedInvokeHash(value, HashChoice<0>{});
         ^
/Users/mntlrndr/temp/borrable_sin_mas/nativescript-plugin-firebase/demo-ng/platforms/ios/Pods/FirebaseFirestore/Firestore/core/src/firebase/firestore/util/hashing.h:136:26: note: in instantiation of
      function template specialization 'firebase::firestore::util::impl::InvokeHash<firebase::firestore::model::ResourcePath>' requested here
  state = Combine(state, InvokeHash(value));
                         ^
/Users/mntlrndr/temp/borrable_sin_mas/nativescript-plugin-firebase/demo-ng/platforms/ios/Pods/FirebaseFirestore/Firestore/core/src/firebase/firestore/util/hashing.h:144:16: note: in instantiation of
      function template specialization 'firebase::firestore::util::impl::HashInternal<firebase::firestore::model::ResourcePath>' requested here
  return impl::HashInternal(0u, values...);
               ^
In file included from /Users/mntlrndr/temp/borrable_sin_mas/nativescript-plugin-firebase/demo-ng/platforms/ios/Pods/FirebaseFirestore/Firestore/core/src/firebase/firestore/model/document.cc:17:
In file included from /Users/mntlrndr/temp/borrable_sin_mas/nativescript-plugin-firebase/demo-ng/platforms/ios/Pods/FirebaseFirestore/Firestore/core/src/firebase/firestore/model/document.h:21:
In file included from /Users/mntlrndr/temp/borrable_sin_mas/nativescript-plugin-firebase/demo-ng/platforms/ios/Pods/FirebaseFirestore/Firestore/core/src/firebase/firestore/model/field_value.h:29:
/Users/mntlrndr/temp/borrable_sin_mas/nativescript-plugin-firebase/demo-ng/platforms/ios/Pods/FirebaseFirestore/Firestore/core/src/firebase/firestore/model/document_key.h:124:18: note: in instantiation
      of function template specialization 'firebase::firestore::util::Hash<firebase::firestore::model::ResourcePath>' requested here
    return util::Hash(key.path());
                 ^
1 error generated.
** ARCHIVE FAILED **
```

any idea?  😕  

**THANKS!**

**Comment by MentalRender** [comment]
Same error on Xcode... 

`/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/functional:2480:5: Static_assert failed "This hash only works for enumeration types"`

![image](https://user-images.githubusercontent.com/17764012/43687669-2f4aebc0-98d9-11e8-8492-03553fe57288.png)

**Comment by EddyVerbruggen** [comment]
Are you on the latest Xcode version?

**Comment by MentalRender** [comment]
Its works!!! Thanks Eddy!

**SOLVED** updating Xcode (9.4.1)
